src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs

```diff
@@ -1,325 +1,56 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-using System;
-using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
-using System.Linq;
 using System.Net.Http;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using System.Web;
-using Microsoft.CodeAnalysis;
-using Microsoft.OpenApi.Models;
 using Microsoft.PowerFx.Connectors.Execution;
-using Microsoft.PowerFx.Core.IR;
-using Microsoft.PowerFx.Core.Types;
-using Microsoft.PowerFx.Core.Utils;
 using Microsoft.PowerFx.Types;
 
 namespace Microsoft.PowerFx.Connectors
 {
     // Given Power Fx arguments, translate into a HttpRequestMessage and invoke.
-    internal class HttpFunctionInvoker
-    {
-        private readonly HttpMessageInvoker _httpClient;
-        private readonly ConnectorFunction _function;
-        private readonly bool _returnRawResults;
-        private readonly ConnectorLogger _logger;
-
-        public HttpFunctionInvoker(ConnectorFunction function, BaseRuntimeConnectorContext runtimeContext)
-        {
-            _function = function;
-            _httpClient = runtimeContext.GetInvoker(function.Namespace);
-            _returnRawResults = runtimeContext.ReturnRawResults;
-            _logger = runtimeContext.ExecutionLogger;
+    internal class HttpFunctionInvoker : FunctionInvoker
```

Comment:
>HttpFunctionInvoker [](http://example.com/codeflow?start=19&length=19)

where do we allocate this?

Reply:
BaseRuntimeConnectorContext.GetInvoker is where we should have this constructor
Should be virtual and not abstract